### PR TITLE
Add QQ Mail, 163 Mail (BROKEN) and Naver Mail, with minor fix to iCloud Mail.

### PR DESCRIPTION
**Pull request by yhh2021**

### CODE REVIEW

File: _providers/qq.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+- qq.com
+- foxmail.com
+server:
+  - type: imap
+    socket: SSL
+    hostname: imap.qq.com
+    port: 993
+    username_pattern: emaillocalpart
+  - type: smtp
+    socket: SSL/TLS
```
Comment (r10s) [inline _providers/qq.md:14]:
```suggestion
    socket: STARTTLS
```

same here.

**Review by r10s — COMMENTED** [review]
thanks a lot for adding asian (?) providers!

i cannot really test them, however, apart of the hints below, code-wise it looks pretty good!

**Comment by yhh2021** [comment]
> thanks a lot for adding asian (?) providers!
> 
> i cannot really test them, however, apart of the hints below, code-wise it looks pretty good!

Thanks for the clues!

QQ Mail and 163 Mail are popular in China's mainland, where over 95% of E-mail users use QQ Mail, 163 Mail, Foxmail and/or 126 Mail, and over 80% use QQ Mail and/or 163 Mail. Foxmail and 126 Mail are respectively the sister brand of QQ Mail and 163 Mail. Naver is the Korean version of Google and Naver Mail is well-liked in Korea.

Earlier today I have tested QQ Mail, 163 Mail and Naver Mail. QQ Mail and Naver Mail works perfectly, but 163 Mail just refuses to connect. I consulted them and was told that they will not this rule, and advised to use their official email client instead. Many bloggers suspect that this measure is set to force 163 Mail users to download 163's own email client. Anyway, seems that currently it is impossible to use Delta Chat with 163 Mail without providing an `IMAP ID' by modifying relevant code as proposed in [163's official documentation]((https://help.mail.163.com/faqDetail.do?code=d7a5dc8471cd0c0e8b4b8f4f8e49998b374173cfe9171305fa1ce630d7f67ac211b1978002df8b23).

**Comment by r10s** [comment]
thanks a lot for these insights and for taking care!

this is very appreciated.

**Review by r10s — APPROVED** [review]

**Pull request merged**